notebooks/.devcontainer/Dockerfile

```diff
@@ -15,7 +15,7 @@ USER $USERNAME
 
 # make all python tools installed by pip accesible
 ENV PATH=$PATH:/home/$USERNAME/.local/bin
-RUN pip install pip --upgrade
+RUN pip install --no-cache-dir pip --upgrade
```

Comment:
just for my understanding, why do we need --no-cache-dir?

Reply:
To reduce the final container image size. check this link out for more details https://stackoverflow.com/questions/45594707/what-is-pips-no-cache-dir-good-for